spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -176,37 +185,29 @@ public Object execute(Object[] parameters) {
 	}
 
 	private Object execute(Object[] parameters, Class returnedElementType, Class<?> collectionType, boolean total) {
-		Supplier<StructuredQuery.Builder<?>> queryBuilderSupplier = StructuredQuery::newKeyQueryBuilder;
-		Function<T, ?> mapper = Function.identity();
-
 		boolean returnedTypeIsNumber = Number.class.isAssignableFrom(returnedElementType)
 				|| returnedElementType == int.class || returnedElementType == long.class;
 
 		boolean isCountingQuery = this.tree.isCountProjection()
 				|| (this.tree.isDelete() && returnedTypeIsNumber) || total;
 
-		Collector<?, ?, ?> collector = Collectors.toList();
-		if (isCountingQuery && !this.tree.isDelete()) {
-			collector = Collectors.counting();
-		}
-		else if (this.tree.isExistsProjection()) {
-			collector = Collectors.collectingAndThen(Collectors.counting(), (count) -> count > 0);
-		}
-		else if (!returnedTypeIsNumber) {
-			queryBuilderSupplier = StructuredQuery::newEntityQueryBuilder;
-			mapper = this::processRawObjectForProjection;
-		}
+		StructuredQuery.Builder<?> structuredQueryBuilder =
+				!((isCountingQuery && !this.tree.isDelete()) || this.tree.isExistsProjection())
```

Comment:
There's way too much conditional logic in this method and many fragments are repeated.
I think it needs to be refactored to be more readable and maintainable.
We can do it in a meeting, if you want.